### PR TITLE
Added support for automatically saving and opening recent folder

### PR DESCRIPTION
**Pull request by m0nk3ydluffy**
This pull request closes issue #1147 
**Before:** Each time opening a file would take you to the home directory
**After:** Now each time you try to open a file, you will land on the directory from where you opened the previous file (or to the default home directory in case this is your first time opening a file)

### CODE REVIEW

File: src/dialogs/NewFileDialog.cpp [hunk 1]
```diff
@@ -84,11 +84,14 @@ void NewFileDialog::on_loadFileButton_clicked()
 
 void NewFileDialog::on_selectFileButton_clicked()
 {
-    const QString &fileName = QDir::toNativeSeparators(QFileDialog::getOpenFileName(this, tr("Select file"), QDir::homePath()));
+    QSettings settings;
```
Comment (xarkes) [inline src/dialogs/NewFileDialog.cpp]: You should use the Cutter configuration singleton. It is available through the `Config()` macro.
You can git grep on that to see how it is used in other places.

File: src/dialogs/OpenFileDialog.cpp [hunk 2]
```diff
@@ -14,10 +15,13 @@ OpenFileDialog::~OpenFileDialog() {}
 
 void OpenFileDialog::on_selectFileButton_clicked()
 {
-    QString fileName = QFileDialog::getOpenFileName(this, tr("Select file"), QDir::homePath());
+    QSettings settings;
+    QString currentDir = settings.value("recentFolder", QDir::homePath()).toString();
```
Comment (xarkes) [inline src/dialogs/OpenFileDialog.cpp]: Same comment here, use the CutterConfiguration :)

**Comment by m0nk3ydluffy** [comment]
Hey @xarkes , I made the changes.

**Comment by xarkes** [comment]
That's nice, looks good to me!
Just don't forget the `QSettings settings;` declaration which is now useless :)

**Comment by m0nk3ydluffy** [comment]
Haha! I guess its finally done now.

**Comment by xarkes** [comment]
Cool, thank you :+1:

**Pull request merged**